zulipterminal/ui_tools/boxes.py

```diff
@@ -104,17 +104,22 @@ def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
             if text.startswith(prefix):
                 typeaheads = autocomplete_func(text, prefix)
                 reduced_typeaheads = typeaheads[:num_suggestions]
+                is_truncated = len(reduced_typeaheads) != len(typeaheads)
 
                 if state in range(-len(reduced_typeaheads),
                                   len(reduced_typeaheads)):
-                    return reduced_typeaheads[state]
+                    typeahead = reduced_typeaheads[state][0]
```

Comment:
Would it make sense to unzip the list of tuples here, so you could set typeahead to one element without the `[0]`, and pass the other unzipped list to `set_typeahead_footer` not as a list of tuples?

Reply:
Fixed by refactoring `match_stream`